api/src/main/java/org/openmrs/BaseChangeableOpenmrsData.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import java.util.Date;
+
+/**
+ * Base superclass for all mutable OpenmrsData.
+ * 
+ * @since 2.2
+ */
+public abstract class BaseChangeableOpenmrsData extends BaseOpenmrsData {
+	
```

Comment:
I added these overrides so that these setters/getters don't appear deprecated as an effect from the superclass since eventually the methods will still exist on this class.

Reply:
Make sense.

Reply:
public interface OpenmrsData extends OpenmrsObject, Auditable, Voidable {}
Which more methods will be added in OpenmrsData interface?
Or OpenmrsData interface will implement the methods of Creatable interface which contain getCreator, getDateCreated methods and Changeable interface contain getChangedBy, getDateChanged methods?

Reply:
OpenmrdsData for now will stay the way it is, in future it will implement Creatable instead of Auditable because Data should be immutable by default.